server.py

```diff
@@ -38,8 +31,63 @@ def response_error():
                 message_in += msg
                 if len(msg) < 16:
                     break
-            conn.sendall(response_ok())
+            try:
+                parse_request(message_in)
+                conn.sendall(response_ok())
+            except (NotImplementedError, ValueError, AttributeError) as e:
+                conn.sendall(response_error(e.message))
             conn.close()
-            print message_in
         except KeyboardInterrupt:
             break
+
+
+def response_ok():
+    response = []
+    now = formatdate(usegmt=True)
+    body = "Thank you for the appropriate request."
+    response.append('HTTP/1.1 200 OK')
+    response.append('Date: {}'.format(now))
+    response.append('Content-Type: text/plain')
+    response.append('Content-Length: {}'.format(len(body)))
+    response.append('Connection: close')
+    response.append('')
+    response.append(body)
+    return CRLF.join(response)
+
+
+def response_error(message):
+    codes = {'405': 'Method Not Allowed',
+             '505': 'HTTP Version Not Supported',
+             '400': 'Bad Request'
+             }
+    response = []
+    now = formatdate(usegmt=True)
+    body = (codes[message])
+    response.append('HTTP 1.1 {} {}'.format(message, codes[message]))
+    response.append('Date: {}'.format(now))
+    response.append('Content-Type: text/plain')
+    response.append('Content-Length: {}'.format(len(body)))
+    response.append('Connection: close')
+    response.append('')
+    response.append(body)
+    return CRLF.join(response)
+
+
+def parse_request(request):
+    request = request.split(CRLF)
+    host = False
+    if 'GET' not in request[0]:
+        raise NotImplementedError('405')
+    elif 'HTTP/1.1' not in request[0]:
+        raise ValueError('505')
+    for line in request[1:]:
+        if 'Host:' in line:
+            host = True
+    if not host:
+        raise AttributeError('400')
+    request_line = request[0].split()
```

Comment:
I'd do this up above, and then compare `GET` and `HTTP/1.1` to the expected slots in the tuple that results.  As it stands an HTTP request with the `GETMEASAMMICH` method would be considered correct, right?